content/batch/workflow-advanced.md

```diff
@@ -0,0 +1,171 @@
+              from: "{{tasks.create-chain.outputs.artifacts.chain}}"
+      - name: Bravo
+        dependencies: [Alpha]
+        template: whalesay
+        arguments:
+          parameters: [{name: message, value: Bravo}]
+          artifacts:
+            - name: chain
+              from: "{{tasks.Alpha.outputs.artifacts.chain}}"
+      - name: Charlie
+        dependencies: [Alpha]
+        template: whalesay
+        arguments:
+          parameters: [{name: message, value: Charlie}]
+          artifacts:
+            - name: chain
+              from: "{{tasks.Alpha.outputs.artifacts.chain}}"
+      - name: Delta
+        dependencies: [Bravo]
+        template: whalesay
+        arguments:
+          parameters: [{name: message, value: Delta}]
+          artifacts:
+            - name: chain
+              from: "{{tasks.Bravo.outputs.artifacts.chain}}"
+      - name: Echo
+        dependencies: [Bravo, Charlie]
+        template: whalesay-reduce
+        arguments:
+          parameters: [{name: message, value: Echo}]
+          artifacts:
+            - name: chain-0
+              from: "{{tasks.Bravo.outputs.artifacts.chain}}"
+            - name: chain-1
+              from: "{{tasks.Charlie.outputs.artifacts.chain}}"
+      - name: Foxtrot
+        dependencies: [Charlie]
+        template: whalesay
+        arguments:
+          parameters: [{name: message, value: Foxtrot}]
+          artifacts:
+            - name: chain
+              from: "{{tasks.create-chain.outputs.artifacts.chain}}"
+      - name: Golf
+        dependencies: [Delta, Echo]
+        template: whalesay-reduce
+        arguments:
+          parameters: [{name: message, value: Golf}]
+          artifacts:
+            - name: chain-0
+              from: "{{tasks.Delta.outputs.artifacts.chain}}"
+            - name: chain-1
+              from: "{{tasks.Echo.outputs.artifacts.chain}}"
+      - name: Hotel
+        dependencies: [Echo, Foxtrot]
+        template: whalesay-reduce
+        arguments:
+          parameters: [{name: message, value: Hotel}]
+          artifacts:
+            - name: chain-0
+              from: "{{tasks.Echo.outputs.artifacts.chain}}"
+            - name: chain-1
+              from: "{{tasks.Foxtrot.outputs.artifacts.chain}}"
+```
+
+This workflow uses a [Directed Acyclic Graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph) (DAG) to explicitly define job dependencies. Each job in the workflow calls a `whalesay` template and passes a parameter with a unique name. Some jobs call a `whalesay-reduce` template which accepts multiple artifacts and combines them into a single artifact.
+
+Each job in the workflow pulls the artifact(s) and lists them in the "Chain", then calls `whalesay` for the current job. Each job will then have a list of the previous job dependency chain (list of all jobs that had to complete before current job could run).
+
+Run the workflow.
+
+```bash
+argo submit --watch teardrop.yaml
```

Comment:
I received an error here because of earlier bucket creation issue

 └-⚠ create-chain  teardrop-vcs77-1270658039  3s        failed to save outputs: Bucket name contains invalid characters

Reply:
This time i am getting Access denied error. Did you miss IAM permissions for S3?

Reply:
Added a section to artifact.md for creating/attaching an IAM policy to the worker nodes.